internal/serverless/client.go

```diff
@@ -0,0 +1,330 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package serverless
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/elastic/elastic-package/internal/environment"
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+const projectsAPI = "/api/v1/serverless/projects"
+
+type Client struct {
+	host   string
+	apiKey string
+}
+
+// ClientOption is functional option modifying Serverless API client.
+type ClientOption func(*Client)
+
+var (
+	ServerlessApiKeyEnvironmentVariable = environment.WithElasticPackagePrefix("SERVERLESS_API_KEY")
+	ServerlessHostvironmentVariable     = environment.WithElasticPackagePrefix("SERVERLESS_HOST")
+
+	ErrProjectNotExist = errors.New("project does not exist")
+)
+
+func NewClient(opts ...ClientOption) (*Client, error) {
+	hostEnvName := ServerlessHostvironmentVariable
+	host := os.Getenv(hostEnvName)
+	if host == "" {
```

Comment:
Ok!
Changed to `os.LookupEnv()` method so it can be distinguished if the variable is not set or it is empty.

If variable is set but empty, I raise the error.

Or would it better experience to just check if the string is empty to set the default host URL ?

Reply:
Yes, in this case we probably also want to use the default if the value is empty.